Use the compiler we setup instead of trying to construct it in cmake
with prefixes.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,7 +211,7 @@
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 
-if (NOT DEFINED WITH_COMPILER_TYPE)
+if (DEFINED FOO)
 
 	# This is for a case when you provided the prefix, but you didn't
 	# provide compiler type. This option is in this form predicted to work
@@ -227,7 +227,7 @@
 	set (USING_DEFAULT_COMPILER_PREFIX 1)
 endif()
 
-if (NOT USING_DEFAULT_COMPILER_PREFIX OR DEFINED WITH_COMPILER_PREFIX)
+if (DEFINED FOO)
 	message(STATUS "Handling compiler with PREFIX=${WITH_COMPILER_PREFIX} TYPE=${WITH_COMPILER_TYPE}")
 
 	parse_compiler_type(${WITH_COMPILER_TYPE} COMPILER_TYPE COMPILER_SUFFIX)
